src/prompt-input/internal.tsx

```diff
@@ -0,0 +1,168 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { Ref, useCallback, useEffect, useRef } from 'react';
+import { getBaseProps } from '../internal/base-component';
+import InternalButton from '../button/internal';
+import { fireKeyboardEvent, fireNonCancelableEvent } from '../internal/events';
+import { PromptInputProps } from './interfaces';
+import { useFormFieldContext } from '../internal/context/form-field-context';
+import useForwardFocus from '../internal/hooks/forward-focus';
+import clsx from 'clsx';
+import styles from './styles.css.js';
+import { InternalBaseComponentProps } from '../internal/hooks/use-base-component';
+import { convertAutoComplete } from '../input/utils';
+import { useDensityMode } from '@cloudscape-design/component-toolkit/internal';
+
+export interface InternalPromptInputProps extends PromptInputProps, InternalBaseComponentProps {}
+
+const InternalPromptInput = React.forwardRef(
+  (
+    {
+      value,
+      actionButtonAriaLabel,
+      actionButtonIconName,
+      actionButtonIconUrl,
+      actionButtonIconSvg,
+      actionButtonIconAlt,
+      ariaLabel,
+      autoComplete = true,
```

Comment:
I'm not sure defaulting this to true makes a lot of sense given the sort of context in which this component will be used
```suggestion
      autoComplete,
```

Reply:
Its set to true by default for the input and textarea components, do you think it'll be confusing for people to have it as the opposite here?

Reply:
`autoComplete` makes most sense in a form context (it's e.g. for auto-filling address fields), so given that the PromptInput is designed to be primarily used outside that context it makes sense to be different